### PR TITLE
fix(Gateway): Export interface

### PR DESCRIPTION
**Pull request by sdanialraza**
**Please describe the changes this PR makes and why it should be merged:**
This interface should be exported.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/core/src/Gateway.ts [hunk 1]
```diff
@@ -2,6 +2,11 @@ import type { Awaitable } from '@discordjs/util';
 import type { ManagerShardEventsMap, WebSocketShardEvents } from '@discordjs/ws';
 import type { GatewaySendPayload } from 'discord-api-types/v10';
 
+/**
+ * Gateway-like structure that can be used to interact with an actual WebSocket connection.
+ * You can provide a custom implementation, useful for running a message broker between your app and your gateway,
+ * or you can simply use the {@link https://discord.js.org/docs/packages/ws/stable/WebSocketManager:Class | WebSocketManager}
```
Comment (Qjuh) [inline packages/core/src/Gateway.ts:8]: Can this be changed to `{@link @discordjs/ws#(WebSocketManager:class)}` to get a dynamic link to the respective version used in docs?

Comment (sdanialraza) [reply packages/core/src/Gateway.ts:8]: That's what I initially wanted to have but that doesn't show properly on editors:
image

Comment (didinele) [reply packages/core/src/Gateway.ts:8]: Doesn't matter, docs have priority

Comment (sdanialraza) [reply packages/core/src/Gateway.ts]: Alright, I've changed it

**Review by didinele — APPROVED** [review]

**Review by Jiralite — APPROVED** [review]

**Review by suneettipirneni — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**